libkmod/libkmod-elf.c

```diff
@@ -679,48 +709,30 @@ static int kmod_elf_get_symbols_symtab(const struct kmod_elf *elf,
 			last = i + 1;
 		}
 	}
-	if (strings[i - 1] != '\0')
-		count++;
```

Comment:
You must keep this check in place, but instead of trying to handle this case, just free memory and return -EINVAL or something like that to indicate that this section is not valid.

Reply:
Indeed - misread the `\0` validation we have in `kmod_elf_new()`.

Maybe it would make more sense to have the validation in `kmod_elf_get_section()`,instead of duplicating it in each caller?

Reply:
I'd actually keep it there. The idea is to support as many (weird) modules as possible and if we don't even want to access symbols, we should not fail, since we do these checks to prevent bugs within kmod, not within the kernel. After all it's the kernel's job to make sure that the modules are valid after all.

Reply:
https://github.com/kmod-project/kmod/pull/222 is another example that these sections need different treatments. Going through __versions for such a verification in a general "let's find offset and size" function seems off.

Reply:
Since it will conflict with your PR and need some fixing - as per reviews - I have switched this to a draft.

Reply:
It would be nice to get this in, because I've just noticed that the current code could lead to overflows on 32 bit systems. If the string section contains a sufficiently long string which a lot of symbols point to (i.e. name offset is identical), the `slen` variable can be incremented past its `SIZE_MAX` limit or at least close enough that the size calculation afterwards overflows. This in turn would mean that our memory allocation is too small, subsequently leading to out ouf boundary writes.

Reply:
I think the updated PR should handle your initial comment. Please have another look - thanks in advance.

Reply:
Yes, this works fine. Maybe there are other occurrences in the code base, but nothing that belongs into this PR.